include/struct_decls/struct_save_state_decl.h

```diff
@@ -0,0 +1,6 @@
+#ifndef POKEPLATINUM_STRUCT_SAVE_STATE_DECL_H
+#define POKEPLATINUM_STRUCT_SAVE_STATE_DECL_H
+
+typedef struct SaveState_t SaveState;
```

Comment:
**polish:** Pull this declaration and its definition into the primary header (and then delete this file); we use transparent structs in this project.